core/common/src/geometry/GeometryStream.ts

```diff
@@ -333,6 +334,35 @@ export class GeometryStreamBuilder {
     return true;
   }
 
+  /** Append a series of entries representing a [[TextBlock]] to the [[GeometryStreamProps]] array.
+   * @beta
+   * @extensions
+   * @preview
+   */
+  public appendTextBlock(block: TextBlockGeometryProps): boolean {
```

Comment:
Going to want ElementGeometry.Builder.appendTextBlock too (flatbuffer api).

Reply:
How do I change the color? It wants a GeometryParams, which wants a category Id, which I don't necessarily know.

Reply:
Backend native will handle an invalid category/sub-category and get them from the element (resets to sub-category appearance). 

For this:
  result = this.appendSubCategoryChange(Id64.invalid);
You should be able to do this:
  const params = new GeometryParams(Id64.invalid);
  builder.appendGeometryParamsChange(params);

For this:
  this.geometryStream.push({ appearance: { color: entry.color } });
You should be able to do this:
  const params = new GeometryParams(Id64.invalid);
  params.lineColor = entry.color;
  builder.appendGeometryParamsChange(params);

Reply:
added.